app/controllers/articles_controller.rb

```diff
@@ -6,4 +6,29 @@ def index
     def show
         @article = Article.find(params[:id])
     end
+
+    def new
+        @article = Article.new
+        @categories = Category.all
+    end
+
+    def create
+        @article = Article.new(article_params)
+        @article.update(category: find_category)
```

Comment:
I'm surprised this is working for you. From what I remember, you can not `update` an AR object that is yet not saved. Anyway, this does not look like a good practice. I would probably use `create` instead of `new` and `persisted?` in the conditional below. (if you want to keep the `update`. But another version is to merge the category key-value pair into the params hash. (still using the `create` - `persisted?` approach)

Reply:
Refactored to the create version, we are unsure how to do merge the category key-value pair into the params hash.

Reply:
Try:
```
@article = Article.new(article_params.merge(category: find_category))
```

Reply:
Refactored, thanks